src/vehicle.cpp

```diff
@@ -3752,9 +3762,52 @@ void vehicle::idle(bool on_map) {
 
     if( on_map ) {
         update_time();
+        if(scoop_on){
+            operate_scoop();
+        }
+    }
+}
+void vehicle::operate_scoop()
+{
+    std::vector<int> scoops = all_parts_with_feature( "SCOOP" );
+    auto veh_points = get_points();
+    for( int scoop : scoops ) {
+        const int chance_to_damage_item = 9;
+        int max_pickup_size = parts[scoop].info().size/10;
+        const char *sound_msgs[] = {_("Whirrrr"), _("Ker-chunk"), _("Swish"), _("Cugugugugug")};
+        sounds::sound( global_pos3() + parts[scoop].precalc[0], rng( 20, 35 ), sound_msgs[rng( 0, 3 )] );
+        for( const tripoint &position : veh_points ) {
```

Comment:
If I'm reading this right, it will pull items from anywhere under the vehicle, I agree with what I think Coolthulu said, it needs to be restricted to the 3x3 squares around where the scoop is installed.